src/transaction/log_page_buffer.c

```diff
@@ -6489,7 +6579,10 @@ logpb_checkpoint (THREAD_ENTRY * thread_p)
 
   /* allocate memory space for the transaction descriptors */
   tmp_chkpt.ntrans = log_Gl.trantable.num_assigned_indices;
-  length_all_chkpt_trans = sizeof (*chkpt_trans) * tmp_chkpt.ntrans;
+  // *INDENT-OFF*
+  ul = log_system_tdes::get_size_and_lock (sys_ntrans);
```

Comment:
better to let map_all_tdes lock mutex. chkpt_trans may be reallocated if the number of sysops exceeds its capacity.